### PR TITLE
[TRA-654] add listing module state init into v7 upgrade handler

### PR DESCRIPTION
**Pull request by shrenujb**
### Changelist
Set markets hard cap and default listing vault params for the listing module state in v7 upgrade handler

### Test Plan
Existing tests

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Integrated a new listing module into the upgrade process, enhancing state initialization during upgrades.
	- Added a hard cap on the number of listed markets and established deposit parameters for the listing vault.

- **Bug Fixes**
	- Implemented error handling to ensure proper initialization of the listing module's state.

- **Tests**
	- Introduced checks to validate the listing module's state after upgrades, ensuring correct setup and functionality.
	- Enhanced post-upgrade validation process with specific checks for the listing module.

### CODE REVIEW

**Comment by vincentwschau** [comment]
Add listing module store to the upgrades [here](https://github.com/dydxprotocol/v4-chain/blob/cd2d378dab5273350db4cfd69dd839378c3651d2/protocol/app/upgrades/v7.0.0/constants.go#L13-L20).

**Comment by shrenujb** [comment]
> Add listing module store to the upgrades [here](https://github.com/dydxprotocol/v4-chain/blob/cd2d378dab5273350db4cfd69dd839378c3651d2/protocol/app/upgrades/v7.0.0/constants.go#L13-L20)

The listing module store key was already added in v6. Do I need it again here?

File: protocol/app/upgrades/v7.0.0/upgrade.go [hunk 1]
```diff
@@ -194,12 +197,30 @@ func migrateVaultSharesToMegavaultShares(ctx sdk.Context, k vaultkeeper.Keeper)
 	ctx.Logger().Info("Successfully migrated vault shares to megavault shares")
 }
 
+func initListingModuleState(ctx sdk.Context, listingKeeper listingkeeper.Keeper) {
+	// Set hard cap on listed markets
+	err := listingKeeper.SetMarketsHardCap(ctx, 500)
```
[the remark this answers is not in the document]
Comment (shrenujb) [reply protocol/app/upgrades/v7.0.0/upgrade.go:202]: A default number in the listing module can change down the line so don't want to reference it here. This code is also just gonna run once

[the remark this answers is not in the document]
Comment (vincentwschau) [reply protocol/app/upgrades/v7.0.0/upgrade.go:202]: Add to `constants.go` instead then?

**Review by vincentwschau — APPROVED** [review]

**Review by vincentwschau — APPROVED** [review]

**Comment by shrenujb** [comment]
@mergifyio backport release/protocol/v7.x

**Pull request merged**